test/runtests.jl

```diff
@@ -620,6 +620,44 @@ const PyInt = pyversion < v"3" ? Int : Clonglong
     @test_throws ArgumentError float(pybuiltin("type"))
 end
 
+@testset "deepcopy #757" begin
+    l = py"[1,2,3]"o
+    l2 = deepcopy(l)
+    @test l == l2
+    l2.append(4)
+    @test l != l2
+    @test collect(l2) == [1,2,3,4]
+    @test collect(l) == [1,2,3]
+
+    obj = py"""
+    class C757:
+        def __init__(self, a, b):
+            self.a = a
+            self.b = b
+    """
+    obj = py"C757(C757(1,2), C757(3,4))"o
+    obj2 = deepcopy(obj)
```

Comment:
```suggestion
    obj2 = deepcopy(obj)
    @test PyPtr(obj) != PyPtr(obj2) # make sure a new Python object is created
```

Reply:
Would be good to test that it actually creates a new object, though I guess you did a related test with `l != l2` above.